pkg/download/download.go

```diff
@@ -89,12 +106,9 @@ func Download(ctx context.Context, endpoint string, opts ...Option) (b []byte, e
 		default:
 		}
 
-		b, err = download(req)
-		if err != nil {
-			return retry.ExpectedError(err)
-		}
+		b, err = download(req, dlOpts)
 
-		return nil
+		return err
```

Comment:
tiny nit: should just `return download(req, dlOpts)` work in this case?

Reply:
the reason is that `b` will not be set, function in `Retry` should return only `error`

Reply:
oh, right. It's nested function. I should have expanded more sources :)